rclcpp/src/rclcpp/parameter_service.cpp

```diff
@@ -24,120 +24,103 @@
 using rclcpp::ParameterService;
 
 ParameterService::ParameterService(
-  const rclcpp::Node::SharedPtr node,
+  const std::shared_ptr<rclcpp::node_interfaces::NodeBaseInterface> node_base,
+  const std::shared_ptr<rclcpp::node_interfaces::NodeServicesInterface> node_services,
+  rclcpp::node_interfaces::NodeParametersInterface * node_params,
   const rmw_qos_profile_t & qos_profile)
-: node_(node)
 {
-  std::weak_ptr<rclcpp::Node> captured_node = node_;
-  get_parameters_service_ = node_->create_service<rcl_interfaces::srv::GetParameters>(
-    std::string(node_->get_name()) + "/" + parameter_service_names::get_parameters,
-    [captured_node](
+  const std::string node_name = node_base->get_name();
+
+  get_parameters_service_ = create_service<rcl_interfaces::srv::GetParameters>(
+    node_base, node_services,
+    std::string(node_name) + "/" + parameter_service_names::get_parameters,
```

Comment:
`node_name` is already a string? Or is it because it's `const`?

Reply:
Neither, I didn't pay close enough attention when replacing `node->get_name()` with `node_name` 07c0b032ec3ced256b2c38290215085ce8951d63

`<string>` defines an `operator+` taking a `const` string and a c string  http://en.cppreference.com/w/cpp/string/basic_string/operator%2B